duviz.py

```diff
@@ -246,6 +246,205 @@ def bar(self, label: str, width: int, fill='-', left='[', right=']', small='|',
         return b
 
 
+class ColumnsRenderer(TreeRenderer):
+    """
+    Render a SizeTree with vertical columns.
+    """
+
+    def __init__(
+        self,
+        height: int,
+        max_depth: int = 5,
+        size_formatter: SizeFormatter = SIZE_FORMATTER_COUNT,
+        color_mode: bool = False,
+    ):
+        super().__init__(max_depth, size_formatter)
+        self.height = height
+        self.color_mode = color_mode
+        if color_mode:
+            self.colorizers = []
+            c = Colorizer()
+            # We use reverse order so that the root is always the same color regardless of the depth
+            for i in reversed(range(max_depth + 1)):
+                cycle = itertools.cycle(
+                    c.COLOR_CYCLE_BMC if i % 2 == 0 else c.COLOR_CYCLE_RGY
+                )
+                self.colorizers.append(cycle)
+
+    def render(self, tree: SizeTree, width: int) -> List[str]:
+        lines = self._render(
+            tree, width // (self.max_depth + 1), self.height, self.max_depth
+        )
+        if lines and not self.color_mode:
+            # The monochrome rendering must be displayed in reverse mode
+            lines[0] = "\x1b[7m" + lines[0]
+            lines[-1] += "\x1b[27m"
+        return lines
+
+    def render_node(
+        self,
+        node: Optional[SizeTree],
+        width: int,
+        height: int,
+        depth: int,
+    ) -> List[str]:
+        """Render a single node"""
+        def _truncate(s, maxlen):
+            truncation_indicator = "..."
+            if len(s) <= maxlen:
+                return s
+            return s[: maxlen - len(truncation_indicator)] + truncation_indicator
+
+        if self.color_mode:
+            color = next(self.colorizers[depth])
+            prefix = color
+            suffix = Colorizer.COLOR_RESET
+            left = ""
+            border_lines = 0
+        else:
+            prefix = ""
+            suffix = ""
+            left = "█"
+            border_lines = 1
+        assert height >= border_lines
+        if node:
+            text_lines = min(2, height - border_lines)
+            size_str = self._size_formatter.format(node.size)
+            if text_lines == 0:
+                label_line_arr = []
+            elif text_lines == 1:
+                name_size_separator = " "
+                name_maxlen = (
+                    width - len(size_str) - len(left) - len(name_size_separator)
+                )
+                label_line_arr = [
+                    _truncate(node.name, name_maxlen) + name_size_separator + size_str
+                ]
+            elif text_lines == 2:
+                name_maxlen = width - len(left)
+                label_line_arr = [
+                    _truncate(node.name, name_maxlen),
+                    size_str,
+                ]
+            else:
+                assert False
```

Comment:
`assert False` will not give a very helpful error message. I think this could at least be `raise ValueError(text_lines)`

Reply:
Done